### PR TITLE
Change Quick Bar shortcuts to "e" and "c"

### PR DESCRIPTION
**Pull request by donkawechico**
## Proposed change

Change quick bar shortcuts:

Quick Bar Mode | Old Shortcut | New shortcut | Short for
------------ | ------------- | ------------- | -------------
Entity Filter | Ctrl+P | **e** | **e**ntity filter
Command List | Ctrl+Shift+P | **c** | **c**ommand filter

**Notes**
1. This should not be merged without https://github.com/home-assistant/frontend/pull/7495 (adds a disable toggle). Single-letter shortcuts pose accessibility risks (see: https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html). For example:
   1. Speech input users can accidentally run a flurry of shortcuts by speaking a word that contains shortcut letters: https://www.youtube.com/watch?v=xzSyIA4OWYE
   2. Users with virtual keyboards can have difficulty pressing the two characters within the timeout
2. This will also affect non-disabled users who use keyboard to navigate to areas of a page quickly

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/15438

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/state/quick-bar-mixin.ts [hunk 1]
```diff
@@ -23,22 +24,50 @@ export default <T extends Constructor<HassElement>>(superClass: T) =>
 
     private _registerShortcut() {
       document.addEventListener("keydown", (e: KeyboardEvent) => {
-        if (!this.hass?.user?.is_admin) {
+        const currentTime = +new Date();
```
Comment (bramkragten) [inline src/state/quick-bar-mixin.ts:27]:
```suggestion
        const currentTime = new Date().getTime();
```

File: src/state/quick-bar-mixin.ts [hunk 2]
```diff
@@ -23,22 +24,50 @@ export default <T extends Constructor<HassElement>>(superClass: T) =>
...
+              this._showQuickBar(e, false);
+              break;
+            case "KeyC":
+              this._showQuickBar(e, true);
+              break;
+            default:
+              this.firstKeyPressedAt = undefined;
+          }
+        } else {
+          this.firstKeyPressedAt = e.code === "KeyQ" ? +new Date() : undefined;
```
Comment (bramkragten) [inline src/state/quick-bar-mixin.ts:52]: Tinykeys would be nice now.

File: src/state/quick-bar-mixin.ts (hunk 1 above)
Comment (donkawechico) [reply src/state/quick-bar-mixin.ts]: Unnecessary now with `tinykeys`

File: src/state/quick-bar-mixin.ts (hunk 2 above)
Comment (donkawechico) [reply src/state/quick-bar-mixin.ts]: Done

File: src/state/quick-bar-mixin.ts [hunk 3]
```diff
@@ -4,15 +4,14 @@ import {
   QuickBarParams,
   showQuickBar,
 } from "../dialogs/quick-bar/show-dialog-quick-bar";
+import tinykeys from "tinykeys";
```
Comment (zsarnett) [inline src/state/quick-bar-mixin.ts:7]: move to the top of the file with other absolute imports

**Review by zsarnett — APPROVED** [review]

File: src/state/quick-bar-mixin.ts [hunk 4]
```diff
@@ -22,23 +21,23 @@ export default <T extends Constructor<HassElement>>(superClass: T) =>
...
-      return isMacOS ? e.metaKey : e.ctrlKey;
+    private _showQuickBar(e: KeyboardEvent, commandMode = false) {
+      if (!this.hass?.user?.is_admin || this.inInputField(e)) {
+        return;
+      }
+
+      showQuickBar(this, { commandMode });
+    }
+
+    private inInputField(e: KeyboardEvent) {
```
Comment (zsarnett) [inline src/state/quick-bar-mixin.ts:38]:
```suggestion
    private _inInputField(e: KeyboardEvent) {
```

File: src/state/quick-bar-mixin.ts [hunk 5]
```diff
@@ -22,23 +21,23 @@ export default <T extends Constructor<HassElement>>(superClass: T) =>
...
+      tinykeys(window, {
+        e: (ev) => this._showQuickBar(ev),
+        c: (ev) => this._showQuickBar(ev, true),
       });
     }
 
-    private isOSCtrlKey(e: KeyboardEvent) {
-      return isMacOS ? e.metaKey : e.ctrlKey;
+    private _showQuickBar(e: KeyboardEvent, commandMode = false) {
+      if (!this.hass?.user?.is_admin || this.inInputField(e)) {
```
Comment (zsarnett) [inline src/state/quick-bar-mixin.ts:31]:
```suggestion
      if (!this.hass?.user?.is_admin || this._inInputField(e)) {
```

**Review by bramkragten — APPROVED** [review]

**Comment by donkawechico** [comment]
We'll also need to update the docs: https://rc.home-assistant.io/docs/tools/quick-bar/

File: src/state/quick-bar-mixin.ts (hunk 5 above)
Comment (donkawechico) [reply src/state/quick-bar-mixin.ts]: done

File: src/state/quick-bar-mixin.ts (hunk 4 above)
Comment (donkawechico) [reply src/state/quick-bar-mixin.ts]: done

File: src/state/quick-bar-mixin.ts (hunk 3 above)
Comment (donkawechico) [reply src/state/quick-bar-mixin.ts]: done

**Comment by donkawechico** [comment]
Documentation update here: https://github.com/home-assistant/home-assistant.io/pull/15438

**Pull request merged**